test/src/server/routes/collection.js

```diff
@@ -0,0 +1,221 @@
+		const data = {
+			collaborators: [collaborator.toJSON()],
+			description: 'some description',
+			entityType: 'Author',
+			name: 'collectionName',
+			privacy: 'public'
+		};
+		const res = await agent.post('/collection/create/handler').send(data);
+		const collection = await new UserCollection({id: res.body.id}).fetch({withRelated: ['collaborators']});
+		collectionJSON = collection.toJSON();
+	});
+	afterEach((done) => {
+		// Clear DB tables then close superagent server
+		truncateEntities().then(() => {
+			agent.close(done);
+		});
+	});
+
+	it('should update the collection and return 200 status code', async () => {
+		const newCollaborator = await createEditor();
+		const newData = {
+			collaborators: [newCollaborator.toJSON()],
+			description: 'new description',
+			entityType: 'Edition',
+			name: 'new collection name',
+			privacy: 'private'
+		};
+
+		const res = await agent.post(`/collection/${collectionJSON.id}/edit/handler`).send(newData);
+		const updatedCollection = await new UserCollection({id: collectionJSON.id}).fetch({withRelated: ['collaborators']});
+		const updatedCollectionJSON = updatedCollection.toJSON();
+
+		expect(res.status).to.equal(200);
+		expect(updatedCollectionJSON.name).to.equal(newData.name);
+		expect(updatedCollectionJSON.entityType).to.equal(newData.entityType);
+		expect(updatedCollectionJSON.description).to.equal(newData.description);
+		expect(updatedCollectionJSON.collaborators.length).to.equal(1);
+		expect(updatedCollectionJSON.collaborators[0].collaboratorId).to.equal(newCollaborator.get('id'));
+	});
+
+	it('should return status code 500 for incorrect entityType', async () => {
+		const data = {
+			collaborators: [collaborator.toJSON()],
+			description: 'some description',
+			entityType: 'incorrect',
+			name: 'collectionName',
+			privacy: 'public'
+		};
+		const res = await agent.post(`/collection/${collectionJSON.id}/edit/handler`).send(data);
+		expect(res.status).to.equal(500);
+	});
+
+	it('should return status code 500 for empty collection name', async () => {
+		const data = {
+			collaborators: [collaborator.toJSON()],
+			description: 'some description',
+			entityType: 'Author',
+			name: '',
+			privacy: 'public'
+		};
+		const res = await agent.post(`/collection/${collectionJSON.id}/edit/handler`).send(data);
+		expect(res.status).to.equal(500);
+	});
+
+	it('should return status 500 when trying to edit entityType of a non empty collection', async () => {
+		const author = await createAuthor();
+		await new UserCollectionItem({
+			bbid: author.get('bbid'),
+			collectionId: collectionJSON.id
+		}).save(null, {method: 'insert'});
+
+		const data = {
+			collaborators: [collaborator.toJSON()],
+			description: 'some description',
+			entityType: 'Edition',
+			name: 'collectionName',
+			privacy: 'public'
+		};
+		const res = await agent.post(`/collection/${collectionJSON.id}/edit/handler`).send(data);
+		expect(res.status).to.equal(500);
```

Comment:
It would be ideal to check the error message to ensure it's this specific error that is being thrown, otherwise if something else breaks we could have a false negative.
You would `expect(res.body.message).to.equal('Trying to change entityType of a non empty collection')`